solr/core/src/java/org/apache/solr/handler/admin/CoreAdminHandler.java

```diff
@@ -452,6 +454,11 @@ void call() throws Exception {
     }
   }
 
+  @Override
```

Comment:
Yes it's backup API from the `ReplicationHandler` but I assumed from the new proposed v2 format that it goes into CoreAdminHandler as it starts with `/cores`
```
/cores/{coreName}/replication/backups
```
But yes you are right, this should goes inside ReplicationHandler. Also I have not tried it yet, but going via CoreAdminHandler I will not have access to the SolrParams that we configure in solrconfig.xml for ReplicationHandler.
```
<requestHandler name="/replication" class="solr.ReplicationHandler">
  <lst name="leader">
    <str name="replicateAfter">optimize</str>
    <str name="backupAfter">optimize</str>
    <str name="confFiles">schema.xml,stopwords.txt,elevate.xml</str>
  </lst>
  <int name="maxNumberOfBackups">2</int>
  <str name="commitReserveDuration">00:00:10</str>
  <lst name="invariants">
    <str name="maxWriteMBPerSec">16</str>
  </lst>
</requestHandler>
```

Reply:
[Q] Does that mean the v2 API format will also change for '/replication'? i.e. instead of starting with '/cores' Will it starts with '/replication'?
something like that.
```
/replication/{coreName}/backups
```

Reply:
> Yes it's backup API from the ReplicationHandler but I assumed from the new proposed v2 format that it goes into CoreAdminHandler as it starts with /cores

Ah, I see!

The short answer here is that there's no 1:1 mapping between the v2 path and the RequestHandler that the functionality originally came from.  Or vice versa.  There are v1 RequestHandlers (e.g. CollectionsHandler) whose functionality is going to be split across a handful of different paths.  And there are v2 paths whose functionality originally came from a number of different RequestHandlers.

So in terms of registering each v2 API, the convention I've been following has been to register them on whatever RequestHandler the functionality originally came from, regardless of the v2 path.

Great observation though - I'll try to clarify this in the "step-by-step" writeup I put in JIRA.

> I will not have access to the SolrParams that we configure in solrconfig.xml for ReplicationHandler

You've got a really keen eye for all the stuff that doesn't work yet! 😛   You're right - we don't currently have a way to get those requestHandler-level settings forwarded to the corresponding v2 API.  Definitely something I hope to fix sooner rather than later.  I spiked out a solution awhile back but it was pretty hacky.  If you're interested in digging into that some, definitely lmk and we can get a JIRA filed and work on it together.

Reply:
I will look into this and then maybe open a ticket as well. Thank you so much for all the help @gerlowskija. I really appreciate it!

Reply:
moved api to `ReplicationHandler`